File: vortex-metrics/src/session.rs
// SPDX-License-Identifier: Apache-2.0
// SPDX-FileCopyrightText: Copyright the Vortex contributors

//! Session extension for accessing metrics.

use vortex_session::SessionExt;

use crate::VortexMetrics;

/// Extension trait for accessing session metrics.
pub trait MetricsSessionExt: SessionExt {
    /// Return the global session metrics registry.
    fn metrics(&self) -> VortexMetrics {
        self.get::<VortexMetrics>().clone()
    }
}
impl<S: SessionExt> MetricsSessionExt for S {}

File: lang/cpp/tests/session.cpp
// SPDX-License-Identifier: Apache-2.0
// SPDX-FileCopyrightText: Copyright the Vortex contributors
#include <catch2/catch_test_macros.hpp>
#include <vortex/session.hpp>

using namespace vortex;

namespace {
TEST_CASE("Session copy and move", "[session]") {
    const Session s;
    Session other;
    other = s;
    Session moved;
    moved = std::move(other);
    Session other2(std::move(moved));
}
} // namespace
